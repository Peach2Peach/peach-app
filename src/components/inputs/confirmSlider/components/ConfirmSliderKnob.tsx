import { Animated } from 'react-native'
import { IconType } from '../../../../assets/icons'
import tw from '../../../../styles/tailwind'
import Icon from '../../../Icon'
import { getBackgroundColor } from '../helpers/getBackgroundColor'
import { useIsMediumScreen } from '../../../../hooks'

type Props = {
  disabled?: boolean
  pan: Animated.Value
  iconId: IconType
  knobWidth: number
}

export const ConfirmSliderKnob = ({ disabled, pan, iconId, knobWidth }: Props) => {
  const icon = {
    color: tw`text-primary-background`.color,
    size: useIsMediumScreen() ? 19 : 16,
  }

  return (
    <Animated.View
      style={[
        {
          width: knobWidth,
          backgroundColor: disabled ? tw`text-black-4`.color : getBackgroundColor(pan),
        },
        tw`flex flex-row justify-center items-center rounded-full gap-0.5 px-2 py-[6px]`,
        tw.md`px-[6px] py-1 gap-1`,
      ]}
    >
      <Icon id={disabled ? 'slash' : iconId} size={icon.size} color={icon.color} />
      <Icon id="chevronsRight" size={icon.size} color={icon.color} />
    </Animated.View>
  )
}
