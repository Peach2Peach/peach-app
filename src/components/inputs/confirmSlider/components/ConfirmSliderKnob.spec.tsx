import { ConfirmSliderKnob } from './ConfirmSliderKnob'
import { createRenderer } from 'react-test-renderer/shallow'
import { Animated } from 'react-native'

describe('ConfirmSliderKnob', () => {
  const renderer = createRenderer()
  it('renders correctly', () => {
    renderer.render(<ConfirmSliderKnob pan={new Animated.Value(0)} iconId="checkCircle" knobWidth={40} />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
  it('renders when disabled', () => {
    renderer.render(<ConfirmSliderKnob disabled pan={new Animated.Value(0)} iconId="checkCircle" knobWidth={40} />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
  it('renders when slided to the end', () => {
    renderer.render(<ConfirmSliderKnob pan={new Animated.Value(1)} iconId="checkCircle" knobWidth={40} />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
  it('renders when slided to 80%', () => {
    renderer.render(<ConfirmSliderKnob pan={new Animated.Value(0.8)} iconId="checkCircle" knobWidth={40} />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
})
