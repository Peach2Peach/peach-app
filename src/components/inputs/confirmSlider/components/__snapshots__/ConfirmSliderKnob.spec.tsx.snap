// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmSliderKnob renders correctly 1`] = `
<ForwardRef(AnimatedComponentWrapper)
  style={
    [
      {
        "backgroundColor": "rgba(245, 101, 34, 1)",
        "width": 40,
      },
      {
        "alignItems": "center",
        "borderRadius": 9999,
        "display": "flex",
        "flexDirection": "row",
        "gap": 2,
        "justifyContent": "center",
        "paddingBottom": 6,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 6,
      },
      {
        "gap": 4,
        "paddingBottom": 4,
        "paddingLeft": 6,
        "paddingRight": 6,
        "paddingTop": 4,
      },
    ]
  }
>
  <Icon
    color="#FFF9F6"
    id="checkCircle"
    size={19}
  />
  <Icon
    color="#FFF9F6"
    id="chevronsRight"
    size={19}
  />
</ForwardRef(AnimatedComponentWrapper)>
`;

exports[`ConfirmSliderKnob renders when disabled 1`] = `
<ForwardRef(AnimatedComponentWrapper)
  style={
    [
      {
        "backgroundColor": "#BAADA7",
        "width": 40,
      },
      {
        "alignItems": "center",
        "borderRadius": 9999,
        "display": "flex",
        "flexDirection": "row",
        "gap": 2,
        "justifyContent": "center",
        "paddingBottom": 6,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 6,
      },
      {
        "gap": 4,
        "paddingBottom": 4,
        "paddingLeft": 6,
        "paddingRight": 6,
        "paddingTop": 4,
      },
    ]
  }
>
  <Icon
    color="#FFF9F6"
    id="slash"
    size={19}
  />
  <Icon
    color="#FFF9F6"
    id="chevronsRight"
    size={19}
  />
</ForwardRef(AnimatedComponentWrapper)>
`;

exports[`ConfirmSliderKnob renders when slided to 80% 1`] = `
<ForwardRef(AnimatedComponentWrapper)
  style={
    [
      {
        "backgroundColor": "rgba(245, 101, 34, 1)",
        "width": 40,
      },
      {
        "alignItems": "center",
        "borderRadius": 9999,
        "display": "flex",
        "flexDirection": "row",
        "gap": 2,
        "justifyContent": "center",
        "paddingBottom": 6,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 6,
      },
      {
        "gap": 4,
        "paddingBottom": 4,
        "paddingLeft": 6,
        "paddingRight": 6,
        "paddingTop": 4,
      },
    ]
  }
>
  <Icon
    color="#FFF9F6"
    id="checkCircle"
    size={19}
  />
  <Icon
    color="#FFF9F6"
    id="chevronsRight"
    size={19}
  />
</ForwardRef(AnimatedComponentWrapper)>
`;

exports[`ConfirmSliderKnob renders when slided to the end 1`] = `
<ForwardRef(AnimatedComponentWrapper)
  style={
    [
      {
        "backgroundColor": "rgba(5, 168, 90, 1)",
        "width": 40,
      },
      {
        "alignItems": "center",
        "borderRadius": 9999,
        "display": "flex",
        "flexDirection": "row",
        "gap": 2,
        "justifyContent": "center",
        "paddingBottom": 6,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 6,
      },
      {
        "gap": 4,
        "paddingBottom": 4,
        "paddingLeft": 6,
        "paddingRight": 6,
        "paddingTop": 4,
      },
    ]
  }
>
  <Icon
    color="#FFF9F6"
    id="checkCircle"
    size={19}
  />
  <Icon
    color="#FFF9F6"
    id="chevronsRight"
    size={19}
  />
</ForwardRef(AnimatedComponentWrapper)>
`;
